cpp/src/traversal/gunrock_sssp.cu

```diff
@@ -0,0 +1,39 @@
+#include <algorithm>
+#include <cugraph/utilities/error.hpp>
+
+#include <cugraph/graph.hpp>
+
+#include <gunrock/algorithms/sssp.hxx>
+
+namespace cugraph {
+
+namespace E = ::gunrock; // gunrock/essentials
+
+template <typename VT, typename ET, typename WT>
+void gunrock_sssp(GraphCSRView<VT, ET, WT> const &graph,
+          WT *distances,
+          VT *predecessors,
+          VT source_vertex)
+{
+  
+  printf("gunrock_sssp: ok\n");
+  
+  auto G = E::graph::build::from_csr<E::memory_space_t::device, E::graph::view_t::csr>(
+      graph.number_of_vertices,
+      graph.number_of_vertices,
+      graph.number_of_edges,
+      graph.offsets,
+      graph.indices,
+      graph.edge_data   
+  );
+  
+  E::sssp::run(G, source_vertex, distances, predecessors);
+}
+
+// explicit instantiation
+template void gunrock_sssp<int, int, float>(GraphCSRView<int, int, float> const &graph,
+                                    float *distances,
+                                    int *predecessors,
+                                    int source_vertex);
+
+}  // namespace cugraph
```

Comment:
this is the actual API call to essentials -- everything else in this pull is boilerplate wrappers + build stuff